sw/host/opentitanlib/src/transport/chip_whisperer/board.rs

```diff
@@ -19,6 +23,38 @@ pub trait Board {
     const PIN_SW_STRAP2: &'static str;
     const PIN_TAP_STRAP0: &'static str;
     const PIN_TAP_STRAP1: &'static str;
+
+    /// Auto-detect UARTs for this board (if possible).
+    fn auto_detect_uarts(serial_number: &str) -> Result<Vec<String>>;
+}
+
+/// Helper function: return a list of serial port name of the
+/// serial ports exported by the SAM3X with a given serial number.
+/// Ports are returned by increasing interface number, except if
+/// reversed is set to true.
+fn find_sam3x_uarts(
+    ttys: &Vec<UsbTty>,
+    serial_number: &str,
+    reversed: bool,
+) -> Result<Vec<UsbTty>> {
+    let mut ports: Vec<UsbTty> = ttys
+        .iter()
+        .filter(|tty| tty.serial_number == Some(serial_number.to_string()))
+        .map(|tty| tty.clone())
+        .collect();
+    ensure!(
+        ports.len() == 2,
```

Comment:
There is a comment above that explains it: the firmware exposes two UARTs. We could make this board dependent but I feel like this would be overkill at this point and also `reversed` would make little sense with more than 2.

Reply:
Ah I now realized that I moved the explanation to `CW310::auto_detect_uarts`, so it should probably be in this function to explain the 2.